File: X-Clone-Backend/src/main/java/com/xclone/xclone/domain/like/Like.java
package com.xclone.xclone.domain.like;

import jakarta.persistence.*;

import java.sql.Timestamp;

@Entity
@Table(name = "likes")
public class Like {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "liker_id")
    private Integer likerId;

    @Column(name = "post_id")
    private Integer likedPostId;

    public Integer getId() {
        return id;
    }

    @Column(name = "created_at", updatable = false, insertable = false)
    private Timestamp createdAt;

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getLikerId() {
        return likerId;
    }

    public void setLikerId(Integer likerId) {
        this.likerId = likerId;
    }

    public Integer getLikedPostId() {
        return likedPostId;
    }

    public void setLikedPostId(Integer likedPostId) {
        this.likedPostId = likedPostId;
    }

    public Timestamp getCreatedAt() {
        return createdAt;
    }

}

File: X-Clone-Backend-FastAPI/app/models/like.py
from __future__ import annotations

from sqlalchemy import Column, Integer, TIMESTAMP

from ..db import Base


class Like(Base):
    __tablename__ = "likes"

    id = Column(Integer, primary_key=True, autoincrement=True)
    liker_id = Column(Integer, nullable=False)
    post_id = Column(Integer, nullable=False)
    created_at = Column(TIMESTAMP, nullable=True)


